### PR TITLE
Bug 1273231 - Simplify log parsing internals

### PR DESCRIPTION
**Pull request by wlach**
This also removes the technically-publicly-facing job log url
POST web api's, which were never used by anyone (reading job log
url information is still supported)

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1510)

### CODE REVIEW

File: treeherder/webapp/api/job_log_url.py [hunk 1]
```diff
@@ -45,37 +45,3 @@ def list(self, request, project, jm):
...
-            parse_status = request.data["parse_status"]
-        except KeyError:
-            raise ParseError(detail=("The parse_status parameter is mandatory for this endpoint"))
-        jm.update_job_log_url_status(pk, parse_status)
-        obj = jm.get_job_log_url_detail(pk)
-        return Response(obj)
-
-    @detail_route(methods=['post'])
-    @with_jobs
-    def parse(self, request, project, jm, pk=None):
```
Comment (edmorley) [inline treeherder/webapp/api/job_log_url.py]: The `parse` endpoint is used by the UI:
https://github.com/mozilla/treeherder/blob/5c5f24e3b3c794897629a203113a9e67f7f40657/ui/js/models/job_log_url.js#L18

Whether we still need that (bug 1076769) is another matter :-)

File: treeherder/model/derived/jobs.py [hunk 2]
```diff
@@ -1901,7 +1901,7 @@ def schedule_log_parsing(self, log_data, priority="normal"):
             tasks[log_obj["job_guid"]].append({
                 "func_name": func_name,
                 "routing_key": routing_key,
-                "job_log_url": log_obj
+                "job_log_url": log_obj['url']
```
Comment (edmorley) [inline treeherder/model/derived/jobs.py]: Do we need to make this change in a deploy prior to removing support for handling the object in `expand_log_url()`?

Comment (wlach) [inline treeherder/model/derived/jobs.py]: Good point, I think we do.

File: treeherder/webapp/api/job_log_url.py (hunk 1 above)
Comment (wlach) [inline treeherder/webapp/api/job_log_url.py]: Gross. Well, no point in removing it now I guess. I'll put that API back in.

**Pull request merged**